src/core/src/op/roi_align_rotated.cpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/roi_align_rotated.hpp"
+
+#include "itt.hpp"
+
+namespace ov {
+namespace op {
+namespace v14 {
+ROIAlignRotated::ROIAlignRotated(const Output<Node>& input,
+                                 const Output<Node>& rois,
+                                 const Output<Node>& batch_indices,
+                                 const int pooled_h,
+                                 const int pooled_w,
+                                 const int sampling_ratio,
+                                 const float spatial_scale,
+                                 const bool clockwise_mode)
+    : ROIAlignBase{input, rois, batch_indices, pooled_h, pooled_w, sampling_ratio, spatial_scale},
+      m_clockwise_mode{clockwise_mode} {
+    constructor_validate_and_infer_types();
```

Comment:
The validation of op is run twice once when ROIAlignBase and 2nd here.
Is better to not execute validation in base class because:
- to avoid double run of validation.
- with new op version there can be some collisions in validation like for new version some case can correct but for base class not but there is still need to use base to share common part.

One solution could be not override `constructor_validate_and_infer_types` but make it base as protected helper and call it only in wanted operators.

Reply:
That was a mistake. fixed.